src/runtime/write_debug_image.cpp

```diff
@@ -142,13 +181,14 @@ WEAK extern "C" int halide_debug_to_file(void *user_context, const char *filenam
         return halide_error_code_bad_dimensions;
     }
 
-    if (auto result = halide_copy_to_host(user_context, buf);
-        result != halide_error_code_success) {
+    if (auto result = halide_copy_to_host(user_context, buf); result != halide_error_code_success) {
+        halide_error(user_context, "debug_to_file(): halide_copy_to_host failed\n");
```

Comment:
copy_to_host has already called halide_error if it has failed. This happens at the lowest level possible (e.g. in cuda_do_multidimensional_copy) where the most information is available. We shouldn't add a redundant call.

Reply:
This didn't get added -- just reformatted by clang-format. I can take it out but it clearly isn't hurting anything...

Reply:
Oh wait, nm, I see. Yeah, there were other returns without calling halide_error() that I fixed, but this one is redundant, removing.